services/venice-server/src/main/java/com/linkedin/venice/listener/ReadQuotaEnforcementHandler.java

```diff
@@ -175,7 +178,7 @@ public void channelRead0(ChannelHandlerContext ctx, RouterRequest request) {
     ReferenceCountUtil.retain(request);
     ctx.fireChannelRead(request);
     stats.recordAllowed(storeName, rcu);
-    return;
+    stats.recordReadQuotaUsage(storeName, (double) rcu / storageNodeBucket.getStaleTokenCount());
```

Comment:
Wondering how `Total()` gives `usageKPS`. Eg: If we have 1 server for a store and have 10 as quota and if all 10 gets used in a second leading to this counter incremented like: `(1/10)+(1/9)+(1/8)+(1/7)+(1/6) +(1/5)+(1/4)+(1/3)+(1/2) +(1/1)` ends up in `2.93` which is not `10 KPS` which was what happened here right.

Reply:
You are right, it should be a gauge instead. Updated.

Reply:
Good catch, in your scenario we actually want something like `(1/10)+(1/10)+(1/10)...` so it adds up to 1 and not 2.93. Let me fix this.